sharding/indexHashedNodesCoordinator.go

```diff
@@ -685,6 +693,80 @@ func (ihgs *indexHashedNodesCoordinator) IsInterfaceNil() bool {
 	return ihgs == nil
 }
 
+// createSelectors creates the consensus group selectors for each shard
+// Not concurrent safe, needs to be called under mutex
+func (ihgs *indexHashedNodesCoordinator) createSelectors(
+	nodesConfig *epochNodesConfig,
+) (map[uint32]RandomSelector, error) {
+	var err error
+	var weights []uint32
+
+	selectors := make(map[uint32]RandomSelector)
+	// weights for validators are computed according to each validator rating
+	for shard, vList := range nodesConfig.eligibleMap {
+		weights, err = ihgs.nodesPerShardSetter.ValidatorsWeights(vList)
+		if err != nil {
+			return nil, err
+		}
+
+		selectors[shard], err = NewSelectorWRS(weights, ihgs.hasher)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return selectors, nil
+}
+
+// ValidatorsWeights returns the
+func (ihgs *indexHashedNodesCoordinator) ValidatorsWeights(validators []Validator) ([]uint32, error) {
+	weights := make([]uint32, len(validators))
+	for i := range validators {
+		weights[i] = 1
+	}
+
+	return weights, nil
+}
+
+func selectValidators(
+	selector RandomSelector,
+	randomness []byte,
+	consensusSize uint32,
+	eligibleList []Validator,
+) ([]Validator, error) {
+	if check.IfNil(selector) {
+		return nil, ErrNilRandomSelector
+	}
+	if len(randomness) == 0 {
+		return nil, ErrNilRandomness
+	}
+
+	// todo: checks for indexes
+	selectedIndexes, err := selector.Select(randomness, consensusSize)
+	if err != nil {
+		return nil, err
+	}
+
+	consensusGroup := make([]Validator, consensusSize)
+	for i := range consensusGroup {
+		consensusGroup[i] = eligibleList[selectedIndexes[i]]
+	}
+
+	displayValidatorsForRandomness(consensusGroup, randomness)
+
+	return consensusGroup, nil
+}
+
+func displayValidatorsForRandomness(validators []Validator, randomness []byte) {
```

Comment:
you can move this function in another file. this one is already too big

Reply:
moved the display functions to common.go